app/services/notices/setup/fetch-failed-returns-invitations.service.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+/**
+ * Fetches the notifications which notify failed to send.
+ * @module FetchFailedReturnsInvitationsService
+ */
+
+const NotificationModel = require('../../../models/notification.model.js')
+
+/**
+ * Fetches the notifications which notify failed to send.
+ *
+ * The function checks an event for failed return invitation emails.
+ *
+ * This could have been because of a temporary issue or an invalid email address.
+ *
+ * @param {string} eventId - The event id to check.
+ *
+ * @returns {Promise<object>} - an object with an array of failed return log ids and an array of failed licence refs
+ */
+async function go(eventId) {
+  const results = await _query(eventId)
+
+  if (results.length === 0) {
+    return []
```

Comment:
This doesn't match what you have stubbed in the tests, which is why the tests are passing, but might fail in real life, as `CreateAlternateNoticeService` is doing the following when an empty array.

```javascript
const { failedLicenceRefs, failedReturnIds } = await FetchFailedReturnsInvitationsService.go(notice.id)
```

Ideally, the result should have the same format regardless of whether any notifications failed.

If you drop this guard clause, you'd achieve the result because the loop would never kick in.